### PR TITLE
20.11 support 

### PR DESCRIPTION
**Pull request by basvandervlies**
This is based on pull request:
 * https://github.com/PySlurm/pyslurm/pull/178 from @bikerdanny 
 * @martijnkruiten has also contributed to this pull request

Maybe its useful for someone

### CODE REVIEW

**Comment by giovtorres** [comment]
Thank you @basvandervlies for this contribution! We'll take a look at this very soon.

File: BUILD.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+```
+The following steps you could do inside the container.
+
+## Checkout PySlurm
+
+```bash
+cd /root
+git clone https://github.com/bikerdanny/pyslurm.git
+cd pyslurm
+git checkout 20.02.5
```
Comment (TimZaman) [inline BUILD.md:40]: `20.02.5`?

**Comment by itkovian** [comment]
Hi, I'm getting an error when running https://github.com/PySlurm/pyslurm/blob/0d5e23aeb85797795d907641479b411626a7739b/examples/listdb_reservations.py

Not sure if this is caused by pyslurm, but we're seeing

~~~~
start=1617798795.14, end=1620390795.14
python: : Unknown error -87664301
~~~~

gdb shows the following

~~~~
(gdb) next
330	in slurm_accounting_storage.c
(gdb) bt
#0  slurm_acct_storage_init () at slurm_accounting_storage.c:330
#1  0x00007fffef963062 in acct_storage_g_get_reservations (db_conn=db_conn@entry=0x0, uid=2005, resv_cond=resv_cond@entry=0x6d6370) at slurm_accounting_storage.c:745
#2  0x00007fffef8eda70 in slurmdb_reservations_get (db_conn=0x0, resv_cond=0x6d6370) at extra_get_functions.c:107
#3  0x00007fffefcb9e21 in __pyx_pw_7pyslurm_7pyslurm_20slurmdb_reservations_7get () from ./pyslurm.so
#4  0x00007ffff7af408c in PyEval_EvalFrameEx () from /lib64/libpython2.7.so.1.0
#5  0x00007ffff7af664d in PyEval_EvalCodeEx () from /lib64/libpython2.7.so.1.0
#6  0x00007ffff7af6752 in PyEval_EvalCode () from /lib64/libpython2.7.so.1.0
#7  0x00007ffff7b0fb8f in run_mod () from /lib64/libpython2.7.so.1.0
#8  0x00007ffff7b11c50 in PyRun_InteractiveOneFlags () from /lib64/libpython2.7.so.1.0
#9  0x00007ffff7b11e3e in PyRun_InteractiveLoopFlags () from /lib64/libpython2.7.so.1.0
#10 0x00007ffff7b124ce in PyRun_AnyFileExFlags () from /lib64/libpython2.7.so.1.0
#11 0x00007ffff7b2319f in Py_Main () from /lib64/libpython2.7.so.1.0
#12 0x00007ffff6d3e555 in __libc_start_main () from /lib64/libc.so.6
#13 0x000000000040068e in ?? ()
#14 0x00007fffffffe2e8 in ?? ()
#15 0x000000000000001c in ?? ()
#16 0x0000000000000001 in ?? ()
#17 0x00007fffffffe592 in ?? ()
#18 0x0000000000000000 in ?? ()
(gdb) next
/usr/bin/python: : Unknown error -274634413
[Inferior 1 (process 27362) exited with code 0210]
~~~~

Offending code:

~~~~
 extern int slurm_acct_storage_init(void)
 {
     int retval = SLURM_SUCCESS;
     char *plugin_type = "accounting_storage";

     if (init_run && plugin_context)
         return retval;

     slurm_mutex_lock(&plugin_context_lock);

     if (plugin_context)
         goto done;

     plugin_context = plugin_context_create(
         plugin_type, slurm_conf.accounting_storage_type, (void **)&ops,
         syms, sizeof(syms));

     if (!plugin_context) {
         error("cannot create %s context for %s",
               plugin_type, slurm_conf.accounting_storage_type);
         retval = SLURM_ERROR;
         goto done;
     }
~~~~



This was with slurm 20.11.6.

**Comment by elgalu** [comment]
Hi! any updates on supporting 20.11?

File: BUILD.md (hunk 1 above)
Comment (basvandervlies) [reply BUILD.md:40]: this has to be adjusted it is not an official release so I can adjust it

**Comment by basvandervlies** [comment]
> Hi, I'm getting an error when running https://github.com/PySlurm/pyslurm/blob/0d5e23aeb85797795d907641479b411626a7739b/examples/listdb_reservations.py
> 
> Not sure if this is caused by pyslurm, but we're seeing
> 
> ```
> start=1617798795.14, end=1620390795.14
> python: : Unknown error -87664301
> ```
> 
> gdb shows the following
> 
> ```
> (gdb) next
> 330	in slurm_accounting_storage.c
> (gdb) bt
> #0  slurm_acct_storage_init () at slurm_accounting_storage.c:330
> #1  0x00007fffef963062 in acct_storage_g_get_reservations (db_conn=db_conn@entry=0x0, uid=2005, resv_cond=resv_cond@entry=0x6d6370) at slurm_accounting_storage.c:745
> #2  0x00007fffef8eda70 in slurmdb_reservations_get (db_conn=0x0, resv_cond=0x6d6370) at extra_get_functions.c:107
> #3  0x00007fffefcb9e21 in __pyx_pw_7pyslurm_7pyslurm_20slurmdb_reservations_7get () from ./pyslurm.so
> #4  0x00007ffff7af408c in PyEval_EvalFrameEx () from /lib64/libpython2.7.so.1.0
> #5  0x00007ffff7af664d in PyEval_EvalCodeEx () from /lib64/libpython2.7.so.1.0
> #6  0x00007ffff7af6752 in PyEval_EvalCode () from /lib64/libpython2.7.so.1.0
> #7  0x00007ffff7b0fb8f in run_mod () from /lib64/libpython2.7.so.1.0
> #8  0x00007ffff7b11c50 in PyRun_InteractiveOneFlags () from /lib64/libpython2.7.so.1.0
> #9  0x00007ffff7b11e3e in PyRun_InteractiveLoopFlags () from /lib64/libpython2.7.so.1.0
> #10 0x00007ffff7b124ce in PyRun_AnyFileExFlags () from /lib64/libpython2.7.so.1.0
> #11 0x00007ffff7b2319f in Py_Main () from /lib64/libpython2.7.so.1.0
> #12 0x00007ffff6d3e555 in __libc_start_main () from /lib64/libc.so.6
> #13 0x000000000040068e in ?? ()
> #14 0x00007fffffffe2e8 in ?? ()
> #15 0x000000000000001c in ?? ()
> #16 0x0000000000000001 in ?? ()
> #17 0x00007fffffffe592 in ?? ()
> #18 0x0000000000000000 in ?? ()
> (gdb) next
> /usr/bin/python: : Unknown error -274634413
> [Inferior 1 (process 27362) exited with code 0210]
> ```
> 
> Offending code:
> 
> ```
>  extern int slurm_acct_storage_init(void)
>  {
>      int retval = SLURM_SUCCESS;
>      char *plugin_type = "accounting_storage";
> 
>      if (init_run && plugin_context)
>          return retval;
> 
>      slurm_mutex_lock(&plugin_context_lock);
> 
>      if (plugin_context)
>          goto done;
> 
>      plugin_context = plugin_context_create(
>          plugin_type, slurm_conf.accounting_storage_type, (void **)&ops,
>          syms, sizeof(syms));
> 
>      if (!plugin_context) {
>          error("cannot create %s context for %s",
>                plugin_type, slurm_conf.accounting_storage_type);
>          retval = SLURM_ERROR;
>          goto done;
>      }
> ```
> 
> This was with slurm 20.11.6.

I did not test this for us the basics works so I expected something has been chamged in the reserveration code.

**Comment by giovtorres** [comment]
@basvandervlies Did this branch ever compile successfully? I pulled down your branch and I think autopxd is missing many declarations. I created a branch based off this PR: https://github.com/PySlurm/pyslurm/tree/basvandervlies-20.11

It doesn't compile yet, but I'll keep working on it. 

/cc @tazend 

Let me know if you would like push access to this branch.

**Comment by basvandervlies** [comment]
@giovtorres yes this branch compiled successfully other people also used this and reported that they could compile and run it. Maybe other things are changes it has been a while ago when I created this patch.  We use this code on our cluster where 20.11 is installed .

**Comment by tazend** [comment]
@giovtorres Yes I can also have a look at it and prepare it in the same way like the 20.2 release

**Comment by giovtorres** [comment]
Closed by #193

**Pull request closed without merging**